File: app/src/main/java/org/secuso/privacyfriendlyrockpaperscissorsboardgame/activities/HomeActivity.java
package org.secuso.privacyfriendlyrockpaperscissorsboardgame.activities;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import org.secuso.privacyfriendlyrockpaperscissorsboardgame.R;
import org.secuso.privacyfriendlyrockpaperscissorsboardgame.core.GameController;

public class HomeActivity extends BaseActivity {
    private ViewPager mViewPager;
    private ImageView mArrowLeft;
    private ImageView mArrowRight;
    public static String GAMEMODE_EXTRA = "MODE";
    public static String AI_EXTRA = "AI";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        PreferenceManager.setDefaultValues(this,R.xml.pref_general,false);
        setContentView(R.layout.activity_home);
        final SectionsPagerAdapter mSectionsPagerAdapter = new SectionsPagerAdapter(getSupportFragmentManager());
        // Set up the ViewPager with the sections adapter.
        mViewPager = (ViewPager) findViewById(R.id.scroller);
        if (mViewPager != null) {
            mViewPager.setAdapter(mSectionsPagerAdapter);
        }

        int index = mSharedPreferences.getInt("lastChosenPage", 0);
        boolean welcomeScreenShown = mSharedPreferences.getBoolean("welcomeScreen",false);
        if(!welcomeScreenShown){
            AlertDialog.Builder builder = new AlertDialog.Builder(this);
            builder.setTitle(R.string.sWelcomeTitle);
            builder.setMessage(R.string.sWelcome);
            builder.setPositiveButton(R.string.sDialogHandOverOkButton, new DialogInterface.OnClickListener() {
                @Override
                public void onClick(DialogInterface dialogInterface, int i) {
                    mSharedPreferences.edit().putBoolean("welcomeScreen",true).apply();
                }
            });
            builder.setNegativeButton(R.string.sHelp, new DialogInterface.OnClickListener() {
                @Override
                public void onClick(DialogInterface dialogInterface, int i) {
                    goToNavigationItem(R.id.nav_help);
                }
            });
            Dialog dialog=builder.create();
            dialog.setOnCancelListener(new DialogInterface.OnCancelListener() {
                @Override
                public void onCancel(DialogInterface dialogInterface) {
                    mSharedPreferences.edit().putBoolean("welcomeScreen",true).apply();
                }
            });
            dialog.show();
        }
        mViewPager.setCurrentItem(index);
        mArrowLeft = (ImageView) findViewById(R.id.arrow_left);
        mArrowRight = (ImageView) findViewById(R.id.arrow_right);

        //care for initial postiton of the ViewPager
        mArrowLeft.setVisibility((index == 0) ? View.INVISIBLE : View.VISIBLE);
        mArrowRight.setVisibility((index == mSectionsPagerAdapter.getCount() - 1) ? View.INVISIBLE : View.VISIBLE);
        ((TextView) HomeActivity.this.findViewById(R.id.GameDesc)).setText(R.string.sGameDescDefault);
        //Update ViewPager on change
        mViewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                mArrowLeft.setVisibility((position == 0) ? View.INVISIBLE : View.VISIBLE);
                mArrowRight.setVisibility((position == mSectionsPagerAdapter.getCount() - 1) ? View.INVISIBLE : View.VISIBLE);
                TextView gameDescription = (TextView) HomeActivity.this.findViewById(R.id.GameDesc);
                switch (position) {
                    case GameController.MODE_NORMAL_AUTO:
                        gameDescription.setText(R.string.sGameDescDefault);
                        break;
                    case GameController.MODE_ROCKPAPERSCISSORSLIZARDSPOCK_AUTO:
                        gameDescription.setText(R.string.sGameDescLizardSpock);
                        break;
                    case GameController.MODE_NORMAL_MANUAL:
                        gameDescription.setText(R.string.sGameDescDefaultManual);
                        break;
                    case GameController.MODE_ROCKPAPERSCISSORSLIZARDSPOCK_MANUAL:
                        gameDescription.setText(R.string.sGameDescLizardSpockManual);
                        break;
                }
                //save position in settings
                SharedPreferences.Editor editor = mSharedPreferences.edit();
                editor.putInt("lastChosenPage", position);
                editor.apply();
            }

            @Override
            public void onPageScrollStateChanged(int state) {
            }
        });

    }

    public void startGame(View view) {
        Intent intent = new Intent(this, GameActivity.class);
        intent.putExtra(AI_EXTRA, false);
        intent.putExtra(GAMEMODE_EXTRA, this.mViewPager.getCurrentItem());
        startActivity(intent);

    }

    public void startGameAI(View view) {
        Intent intent = new Intent(this, GameActivity.class);
        intent.putExtra(AI_EXTRA, true);
        intent.putExtra(GAMEMODE_EXTRA, this.mViewPager.getCurrentItem());
        startActivity(intent);

    }

    public void continueGame(View view){
        Intent intent = new Intent(this,ContinueActivity.class);
        startActivity(intent);
    }

    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.arrow_left:
                mViewPager.arrowScroll(View.FOCUS_LEFT);
                break;
            case R.id.arrow_right:
                mViewPager.arrowScroll(View.FOCUS_RIGHT);
                break;
            default:
        }
    }

    @Override
    protected int getNavigationDrawerID() {
        return R.id.nav_home;
    }

    public class SectionsPagerAdapter extends FragmentPagerAdapter {

        public SectionsPagerAdapter(FragmentManager fm) {
            super(fm);
        }


        @Override
        public Fragment getItem(int position) {
            // getItem is called to instantiate the fragment for the given page.
            // Return a PageFragment (defined as a static inner class below).
            return PageFragment.newInstance(position);
        }

        @Override
        public int getCount() {
            // Show 4 total pages.
            return 4;
        }
    }

    public static class PageFragment extends Fragment {
        /**
         * The fragment argument representing the section number for this
         * fragment.
         */
        private static final String ARG_SECTION_NUMBER = "section_number";

        /**
         * Returns a new instance of this fragment for the given section
         * number.
         */
        public static PageFragment newInstance(int sectionNumber) {
            PageFragment fragment = new PageFragment();
            Bundle args = new Bundle();
            args.putInt(ARG_SECTION_NUMBER, sectionNumber);
            fragment.setArguments(args);
            return fragment;
        }

        public PageFragment() {

        }


        @Override
        public View onCreateView(LayoutInflater inflater, ViewGroup container,
                                 Bundle savedInstanceState) {
            int id = 0;
            if (getArguments() != null) {
                id = getArguments().getInt(ARG_SECTION_NUMBER);
            }

            View rootView = inflater.inflate(R.layout.fragment_main_menu, container, false);

            ImageView imageView = (ImageView) rootView.findViewById(R.id.section_label);
            switch(id){
                case 1:
                    imageView.setBackgroundResource(R.drawable.rpslsfix);
                    break;
                case 2:
                    imageView.setBackgroundResource(R.drawable.rpsman);
                    break;
                case 3:
                    imageView.setBackgroundResource(R.drawable.rpslsman);
                    break;
                default:
                    imageView.setBackgroundResource(R.drawable.rpsfix);
                    break;
            }
            return rootView;
        }
    }
}

File: app/src/main/java/org/secuso/privacyfriendlyrockpaperscissorsboardgame/activities/HomeActivity.kt
package org.secuso.privacyfriendlyrockpaperscissorsboardgame.activities

import android.content.Intent
import android.os.Bundle
import android.preference.PreferenceManager
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.support.v4.view.ViewPager
import android.support.v4.view.ViewPager.OnPageChangeListener
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import org.secuso.privacyfriendlyrockpaperscissorsboardgame.R
import org.secuso.privacyfriendlyrockpaperscissorsboardgame.checkGoodbyeGoogle
import org.secuso.privacyfriendlyrockpaperscissorsboardgame.core.GameController

class HomeActivity : BaseActivity() {
    private var mViewPager: ViewPager? = null
    private var mArrowLeft: ImageView? = null
    private var mArrowRight: ImageView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        PreferenceManager.setDefaultValues(this, R.xml.pref_general, false)
        setContentView(R.layout.activity_home)
        val mSectionsPagerAdapter = SectionsPagerAdapter(supportFragmentManager)
        // Set up the ViewPager with the sections adapter.
        mViewPager = findViewById<View>(R.id.scroller) as ViewPager
        if (mViewPager != null) {
            mViewPager!!.adapter = mSectionsPagerAdapter
        }

        val index = mSharedPreferences!!.getInt("lastChosenPage", 0)

        mViewPager!!.currentItem = index
        mArrowLeft = findViewById<View>(R.id.arrow_left) as ImageView
        mArrowRight = findViewById<View>(R.id.arrow_right) as ImageView

        //care for initial postiton of the ViewPager
        mArrowLeft!!.visibility = if ((index == 0)) View.INVISIBLE else View.VISIBLE
        mArrowRight!!.visibility = if ((index == mSectionsPagerAdapter.count - 1)) View.INVISIBLE else View.VISIBLE
        (findViewById<View>(R.id.GameDesc) as TextView).setText(R.string.sGameDescDefault)
        //Update ViewPager on change
        mViewPager!!.addOnPageChangeListener(object : OnPageChangeListener {
            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
            }

            override fun onPageSelected(position: Int) {
                mArrowLeft!!.visibility = if ((position == 0)) View.INVISIBLE else View.VISIBLE
                mArrowRight!!.visibility = if ((position == mSectionsPagerAdapter.count - 1)) View.INVISIBLE else View.VISIBLE
                val gameDescription = findViewById<View>(R.id.GameDesc) as TextView
                when (position) {
                    GameController.MODE_NORMAL_AUTO -> gameDescription.setText(R.string.sGameDescDefault)
                    GameController.MODE_ROCKPAPERSCISSORSLIZARDSPOCK_AUTO -> gameDescription.setText(R.string.sGameDescLizardSpock)
                    GameController.MODE_NORMAL_MANUAL -> gameDescription.setText(R.string.sGameDescDefaultManual)
                    GameController.MODE_ROCKPAPERSCISSORSLIZARDSPOCK_MANUAL -> gameDescription.setText(R.string.sGameDescLizardSpockManual)
                }
                //save position in settings
                val editor = mSharedPreferences!!.edit()
                editor.putInt("lastChosenPage", position)
                editor.apply()
            }

            override fun onPageScrollStateChanged(state: Int) {
            }
        })
        checkGoodbyeGoogle(this, layoutInflater)
    }

    fun startGame(view: View?) {
        val intent = Intent(this, GameActivity::class.java)
        intent.putExtra(AI_EXTRA, false)
        intent.putExtra(GAMEMODE_EXTRA, mViewPager!!.currentItem)
        startActivity(intent)
    }

    fun startGameAI(view: View?) {
        val intent = Intent(this, GameActivity::class.java)
        intent.putExtra(AI_EXTRA, true)
        intent.putExtra(GAMEMODE_EXTRA, mViewPager!!.currentItem)
        startActivity(intent)
    }

    fun continueGame(view: View?) {
        val intent = Intent(this, ContinueActivity::class.java)
        startActivity(intent)
    }

    fun onClick(view: View) {
        when (view.id) {
            R.id.arrow_left -> mViewPager!!.arrowScroll(View.FOCUS_LEFT)
            R.id.arrow_right -> mViewPager!!.arrowScroll(View.FOCUS_RIGHT)
            else -> {}
        }
    }

    override val navigationDrawerID: Int
        get() = R.id.nav_home

    inner class SectionsPagerAdapter(fm: FragmentManager?) : FragmentPagerAdapter(fm) {
        override fun getItem(position: Int): Fragment {
            // getItem is called to instantiate the fragment for the given page.
            // Return a PageFragment (defined as a static inner class below).
            return PageFragment.newInstance(position)
        }

        override fun getCount(): Int {
            // Show 4 total pages.
            return 4
        }
    }

    class PageFragment : Fragment() {
        override fun onCreateView(
            inflater: LayoutInflater, container: ViewGroup?,
            savedInstanceState: Bundle?
        ): View? {
            var id = 0
            if (arguments != null) {
                id = arguments!!.getInt(ARG_SECTION_NUMBER)
            }

            val rootView = inflater.inflate(R.layout.fragment_main_menu, container, false)

            val imageView = rootView.findViewById<View>(R.id.section_label) as ImageView
            when (id) {
                1 -> imageView.setBackgroundResource(R.mipmap.rpslsfix)
                2 -> imageView.setBackgroundResource(R.mipmap.rpsman)
                3 -> imageView.setBackgroundResource(R.mipmap.rpslsman)
                else -> imageView.setBackgroundResource(R.mipmap.rpsfix)
            }
            return rootView
        }

        companion object {
            /**
             * The fragment argument representing the section number for this
             * fragment.
             */
            private const val ARG_SECTION_NUMBER = "section_number"

            /**
             * Returns a new instance of this fragment for the given section
             * number.
             */
            fun newInstance(sectionNumber: Int): PageFragment {
                val fragment = PageFragment()
                val args = Bundle()
                args.putInt(ARG_SECTION_NUMBER, sectionNumber)
                fragment.arguments = args
                return fragment
            }
        }
    }

    companion object {
        var GAMEMODE_EXTRA: String = "MODE"
        var AI_EXTRA: String = "AI"
    }
}